lib/token-translator.js

```diff
@@ -123,8 +123,9 @@ TokenTranslator.prototype = {
                  type === tt.arrow || type === tt.jsxTagStart ||
                  type === tt.incDec || type === tt.starstar ||
                  type === tt.jsxTagEnd || (type.binop && !type.keyword) ||
-                 type.isAssign) {
-
+                 type.isAssign ||
+                 (type.prefix && (token.value === "~" || token.value === "!"))
```

Comment:
I just checked again: `typeof`, `void` and `delete` are also [prefixes](https://github.com/ternjs/acorn/blob/9349eb4f4ec91da2b7236f2218e84a51bee22969/src/tokentype.js#L144). So a differentiation is necessary.